### ISSUE TITLE
Recursive build support

### ISSUE THREAD

**Issue by tpn** [conda/conda-build#49]
There needs to be a way for conda-build to automatically try and build any missing dependencies if it can match the dependency to a recipe directory.  This functionality is essential for building the R cran-conda-recipes as a given recipe may have anywhere from 5-15 different dependencies that won't have been built yet.

I put in very hacky support in the original r branch via this commit:
https://github.com/conda/conda/commit/efba6b5a47a2e9aef8781a731ff1f48da56e49db

Which I'd then make use of with github.com/tpn/cran-conda-recipes/rbuild:
https://github.com/tpn/cran-conda-recipes/blob/master/rbuild

Basically, I made conda write the name of the missing package to conda-missing-package.txt, then I'd read that after a `conda build` failure, and recursively keep building failed deps until it worked, then re-try failed deps again whilst unwinding the stack.

It was super hacky but it worked.  The elegant solution would be to make the solver cognizant of build requirements, not just install-time requirements -- but I'd be happy with whatever gets me recursive build functionality again, ideally in the master/mainline.
